homeassistant/components/anova/__init__.py

```diff
@@ -17,15 +24,44 @@
 
 async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     """Set up Anova from a config entry."""
-    devices = [
+    api = AnovaApi(
+        aiohttp_client.async_get_clientsession(hass),
+        entry.data[CONF_USERNAME],
+        entry.data[CONF_PASSWORD],
+    )
+    try:
+        await api.authenticate()
+        online_devices = await api.get_devices()
+    except InvalidLogin as err:
+        raise ConfigEntryNotReady("Login was incorrect, please try again") from err
+    except NoDevicesFound:
+        # get_devices raises an exception if no devices are online
+        online_devices = []
```

Comment:
```suggestion
        raise ConfigEntryNotReady("No devices are online") from err
```

Reply:
Could go either way on this one.  Whatever you think is best

Reply:
I think allowing it to continue even if there are no devices online is preferred - as the devices themselves will return UpdateFailed if they are offline and the update() functionality of the device is a lot easier of a call (just a basic rest call) vs the get devices is a more complicated websocket connect and parsing, so if one has to repeat while it is online I think the update() would be better.